src/platforms/common/px4_work_queue/WorkQueueManager.hpp

```diff
@@ -0,0 +1,141 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2019 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include <containers/BlockingQueue.hpp>
+#include <containers/List.hpp>
+#include <containers/Queue.hpp>
+
+#include <px4_defines.h>
+#include <px4_module.h>
+#include <px4_tasks.h>
+
+extern "C" __EXPORT int wq_manager_main(int argc, char *argv[]);
+
+namespace px4
+{
+
+// work queues
+
+// SPI1 : PRIORITY MAX : STACK 750
+
+
+enum PX4_WQS {
+	SPI1 = 0,
+	SPI2,
+
+	I2C1,
+	I2C2,
+
+	rate_ctrl,
+	hp_default,
```

Comment:
Where is `lp_default`?

Reply:
Just hasn't come up yet.

Reply:
Added.

Reply:
Nevermind, I'll need to rethink the lazy relative priorities. The priority ranges are much different between platforms.

Reply:
I'll revisit this once we have our first low priority worker.